include/envoy/thread/thread.h

```diff
@@ -10,16 +12,27 @@
 namespace Envoy {
 namespace Thread {
 
+/**
+ * An id for a thread.
+ */
 class ThreadId {
 public:
-  virtual ~ThreadId() = default;
-
-  virtual std::string debugString() const PURE;
-  virtual bool isCurrentThreadId() const PURE;
+  ThreadId() : id_(std::numeric_limits<int64_t>::min()) {}
+  explicit ThreadId(int64_t id) : id_(id) {}
+  ~ThreadId() = default;
+
+  std::string debugString() const { return std::to_string(id_); }
+  bool isEmpty() const { return *this == ThreadId(); }
+  friend bool operator==(ThreadId lhs, ThreadId rhs) { return lhs.id_ == rhs.id_; }
+  friend bool operator!=(ThreadId lhs, ThreadId rhs) { return lhs.id_ != rhs.id_; }
+  template <typename H> friend H AbslHashValue(H h, ThreadId id) {
+    return H::combine(std::move(h), id.id_);
+  }
+
+private:
+  int64_t id_;
```

Comment:
nit: const

Reply:
I think we don't want this.  As it stands, this is value type that can be copied and assigned normally.  If we make this const we lost assignment